internal/testrunner/runners/system/runner.go

```diff
@@ -73,34 +73,39 @@ type logsByContainer struct {
 	patterns      []logsRegexp
 }
 
-var (
-	errorPatterns = []logsByContainer{
-		logsByContainer{
-			containerName: "elastic-agent",
-			patterns: []logsRegexp{
-				logsRegexp{
-					includes: regexp.MustCompile("^Cannot index event publisher.Event"),
-					excludes: []*regexp.Regexp{
-						// this regex is excluded to ensure that logs coming from the `system` package installed by default are not taken into account
-						regexp.MustCompile(`action \[indices:data\/write\/bulk\[s\]\] is unauthorized for API key id \[.*\] of user \[.*\] on indices \[.*\], this action is granted by the index privileges \[.*\]`),
-					},
+var errorPatterns = []logsByContainer{
+	{
+		containerName: "elastic-agent",
+		patterns: []logsRegexp{
+			{
+				includes: regexp.MustCompile("^Cannot index event publisher.Event"),
+				excludes: []*regexp.Regexp{
+					// this regex is excluded to ensure that logs coming from the `system` package installed by default are not taken into account
+					regexp.MustCompile(`action \[indices:data\/write\/bulk\[s\]\] is unauthorized for API key id \[.*\] of user \[.*\] on indices \[.*\], this action is granted by the index privileges \[.*\]`),
 				},
-				logsRegexp{
-					includes: regexp.MustCompile("->(FAILED|DEGRADED)"),
+			},
+			{
+				includes: regexp.MustCompile("->(FAILED|DEGRADED)"),
 
-					// this regex is excluded to avoid a regresion in 8.11 that can make a component to pass to a degraded state during some seconds after reassigning or removing a policy
-					excludes: []*regexp.Regexp{
-						regexp.MustCompile(`Component state changed .* \(HEALTHY->DEGRADED\): Degraded: pid .* missed .* check-in`),
-					},
+				// this regex is excluded to avoid a regresion in 8.11 that can make a component to pass to a degraded state during some seconds after reassigning or removing a policy
+				excludes: []*regexp.Regexp{
+					regexp.MustCompile(`Component state changed .* \(HEALTHY->DEGRADED\): Degraded: pid .* missed .* check-in`),
 				},
 			},
 		},
-	}
-)
+	},
+}
 
 type runner struct {
 	options   testrunner.TestOptions
 	pipelines []ingest.Pipeline
+
+	dataStreamPath  string
+	cfgFiles        []string
+	variants        []string
+	stackVersion    kibana.VersionInfo
+	locationManager *locations.LocationManager
```

Comment:
The location manager is only used to obtain the path to the services state dir, maybe we could initiallize the runner directly with the path.
```suggestion
	serviceStatePath string
```

Reply:
`locationManager` is also used to create the service log dir for the agent and the output dir:
https://github.com/elastic/elastic-package/blob/49345174cf32ab05dc8f6d57897a225bd51d7539/internal/testrunner/runners/system/runner.go#L284-L294

So I think it could be kept and maybe just add the suggested field as a new variable without removing the locationManager. WDYT ? @jsoriano 

If we do so, that would help us to remove around 9 or 10 calls to `r.locationManager.ServiceSetupDir()`.  

```go
type runner struct {
	options   testrunner.TestOptions
	pipelines []ingest.Pipeline

	dataStreamPath   string
	cfgFiles         []string
	variants         []string
	stackVersion     kibana.VersionInfo
	locationManager  *locations.LocationManager
	serviceStatePath string
    ...
}
```

Reply:
With the latest changes, (moving this file under the profiles directory), I added two new fields here:
https://github.com/mrodm/elastic-package/blob/baad699d966620fa63514ab5178f133ee51fa405/internal/testrunner/runners/system/runner.go#L109-L110
```
	serviceStateFolder   string
	serviceStateFilePath string
```

There are two since it is also required the folder path to create/delete the whole folder.

To use the same paths in cmd/testrunner.go  a new function has been added in the `internal/testrunner/testrunner.go` to try to avoid duplication:

https://github.com/mrodm/elastic-package/blob/baad699d966620fa63514ab5178f133ee51fa405/internal/testrunner/testrunner.go#L329-L331

Reply:
> There are two since it is also required the folder path to create/delete the whole folder.

Cannot we use `filepath.Dir(serviceStateFilePath)` to get the folder?

Reply:
> Cannot we use `filepath.Dir(serviceStateFilePath)` to get the folder?

Sure, I'll update the methods using the folder with that call. No need to keep that field in the struct